tests/backup/backup_helper.go

```diff
@@ -2149,6 +2157,133 @@ func VerifyLicenseConsumedCount(ctx context.Context, OrgId string, expectedLicen
 	return err
 }
 
+// DeleteRule deletes backup rule
+func DeleteRule(ruleName string, orgId string, ctx context.Context) error {
+	ruleUid, err := Inst().Backup.GetRuleUid(orgID, ctx, ruleName)
+	if err != nil {
+		return err
+	}
+	deleteRuleReq := &api.RuleDeleteRequest{
+		OrgId: orgId,
+		Name:  ruleName,
+		Uid:   ruleUid,
+	}
+	_, err = Inst().Backup.DeleteRule(ctx, deleteRuleReq)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SafeAppend appends elements to a given slice in a thread-safe manner using a provided mutex
+func SafeAppend(mu *sync.Mutex, slice interface{}, elements ...interface{}) interface{} {
+	mu.Lock()
+	defer mu.Unlock()
+	sliceValue := reflect.ValueOf(slice)
+	for _, elem := range elements {
+		elemValue := reflect.ValueOf(elem)
+		sliceValue = reflect.Append(sliceValue, elemValue)
+	}
+	return sliceValue.Interface()
+}
```

Comment:
I feel like directly using lock and unlock would be much more performative, and the number of lines of code is less enough to justify not using a separate `SafeAppend` function for that. Like:
```go
m.lock()
s = append(s, ele...)
m.Unlock()
```